pkg/kev/config/svcext.go

```diff
@@ -299,26 +304,18 @@ func getServiceType(serviceType string) (string, error) {
 }
 
 // WorkloadRestartPolicyFromCompose infers a kev-valid restart policy from compose data.
-func WorkloadRestartPolicyFromCompose(svc *composego.ServiceConfig) string {
-	policy := RestartPolicyAlways
-
-	if svc.Restart != "" {
-		policy = svc.Restart
-	}
-
-	if svc.Deploy != nil && svc.Deploy.RestartPolicy != nil {
-		policy = svc.Deploy.RestartPolicy.Condition
-	}
-
-	if policy == "unless-stopped" {
-		log.WarnWithFields(log.Fields{
-			"restart-policy": policy,
-		}, "Restart policy 'unless-stopped' is not supported, converting it to 'always'")
-
-		policy = "always"
+func WorkloadRestartPolicyFromCompose(svc *composego.ServiceConfig) RestartPolicy {
+	switch {
+	case svc.Restart != "": // docker-compose v2
```

Comment:
I don't think it's specific to v2, is it? Also, we should cascade here i.e. the value defined in `Deploy` block should override the value at the top level `Restart`.

Reply:
Nope ... Its a v2 vs v3 thing. No cascade required. Hence the re-implementation.

v2: https://docs.docker.com/compose/compose-file/compose-file-v2/#restart
v3: https://docs.docker.com/compose/compose-file/compose-file-v3/#restart_policy

Reply:
I asked as `Restart` is also possible in [V3 manifest](https://docs.docker.com/compose/compose-file/compose-file-v3/#restart), and thus we should still cascade I think.

Reply:
That's odd the spec states otherwise.

Reply:
Corrected.